### ISSUE TITLE
Catch wrong stateless component function signature

### ISSUE THREAD

**Issue by pke** [yannickcr/eslint-plugin-react#521]
I had this code for my component:

```
const Toggle = (selected, disabled, onChange, children) => (...)
```

The linter should catch the missing `{}` around the properties. It should accept only `one` argument (usually `props`) or an ES6 object splash like this:

```
const Toggle = ({selected, disabled, onChange, children}) => (...)
```

**Comment by yannickcr** [comment]
There is no rules to catch this sort of mistakes right now. Also stateless components can have 2 arguments (`props` and `context`).

However, since detecting stateless components is already very difficult, I would rather consider functions with more than 2 arguments as not being stateless components instead of stateless components with a typo in the arguments.

**Comment by pke** [comment]
Ah you right, forgot about the context arg.
A rule for functions with more than 2 args would have helped me in that case.

**Comment by pke** [comment]
Closing this as there seems to be no demand for it.
